### PR TITLE
386 add levmar

### LINKED ISSUE #386: Add LEVMAR solver support
The `levmar` solver is a Levenberg-Marquardt nonlinear least squares algorithm in C/C++
with python bindings, it would be interesting to compare its performance wrt the other solvers.

It can be downloaded from http://users.ics.forth.gr/~lourakis/levmar/index.html.

### PR DESCRIPTION
**Pull request by tyronerees**
#### Description of Work

Fixes #386 

#### Testing Instructions

1. Choose levmar in the options
2. Check that the results are consistent
3.

Function: Does the change do what it's supposed to?

Tests: Does it pass? Is there adequate coverage for new code?

Style: Is the coding style consistent? Is anything overly confusing?

Documentation: Is there a suitable change to documentation for this change?

### CODE REVIEW

File: fitbenchmarking/controllers/levmar_controller.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+        else:
+            jac = self._jeval
+
+        (self.final_params,
+         _,
+         self._info) = levmar.levmar(self._feval,
+                                     self.initial_params,
+                                     self.data_y,
+                                     args=(self.data_x,),
+                                     jacf=jac)
```
Comment (AndrewLister-STFC) [inline fitbenchmarking/controllers/levmar_controller.py:94]: Formatting suggestion, feel free to ignore.

```suggestion
        self.final_params, _, self._info = levmar.levmar(
            self._feval,
            self.initial_params,
            self.data_y,
            args=(self.data_x,),
            jacf=jac)
```

**Review by AndrewLister-STFC — CHANGES_REQUESTED** [review]
Looks good to me, I think there's 1 link that needs updating and I had a couple of suggestions.

It runs on my laptop and gives answers that make sense.

Can you also include it in the regression tests if it's deterministic please.

File: docs/source/users/install_instructions/fitbenchmarking.rst [hunk 2]
```diff
@@ -77,6 +77,7 @@ where valid strings ``option-x`` are:
 
 * ``bumps``-- installs the `Bumps <https://bumps.readthedocs.io>`_ fitting package.
 * ``DFO`` -- installs the `DFO-LS <http://people.maths.ox.ac.uk/robertsl/dfols/userguide.html>`_ and `DFO-GN <http://people.maths.ox.ac.uk/robertsl/dfogn/userguide.html>`_ fitting packages.
+* ``levmar`` -- installs the `levmar <https://github.com/SasView/sasmodels>`_ fitting package.  Note that the interface we use also requires BLAS and LAPLACK to be installed on the system, and calls to this minimizer will fail if these libraries are not present.
```
[the remark this answers is not in the document]
Comment (tyronerees) [reply docs/source/users/install_instructions/fitbenchmarking.rst:80]:
```suggestion
* ``levmar`` -- installs the `levmar <http://users.ics.forth.gr/~lourakis/levmar/>`_ fitting package.  Note that the interface we use also requires BLAS and LAPLACK to be installed on the system, and calls to this minimizer will fail if these libraries are not present.
```

File: fitbenchmarking/controllers/levmar_controller.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+
+        :param p: parameters
+        :type p: list
+        :param data: x data
+        :type data: list
+        :return: result from problem.eval_model
+        :rtype: numpy array
+        """
+
+        fx = self.problem.function(self.data_x, *p)  # (eval_model(params=p)
```
[the remark this answers is not in the document]
Comment (tyronerees) [reply fitbenchmarking/controllers/levmar_controller.py:62]:
```suggestion
        fx = self.problem.function(self.data_x, *p) 
```

File: fitbenchmarking/controllers/levmar_controller.py [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+        :param p: parameters
+        :type p: list
+        :param data: x data, this is discarded as the defaults can be used.
+        :type data: N/A
+        :return: result from jac.eval
+        :rtype: numpy array
+        """
+        jac = -self.jacobian.eval(p)
+        return jac
```
Comment (jess-farmer) [inline fitbenchmarking/controllers/levmar_controller.py:77]: Pylint throws up an unused argument warning for the x argument in these functions

**Comment by jess-farmer** [comment]
Looks good to me, just thought I would mention the 'unused argument' pylint warning as we're looking to remove all linting errors and warnings

**Pull request merged**